yt_dlp/extractor/theplatform.py

```diff
@@ -295,6 +301,15 @@ def _real_extract(self, url):
 
         formats, subtitles = self._extract_theplatform_smil(smil_url, video_id)
 
+        # With some sites, extracting from manifest URL directly yields more formats
+        m3u8_url = update_url(url, query='mbr=true&manifest=m3u', fragment=None)
+        ext = urlhandle_detect_ext(self._request_webpage(
+            HEADRequest(m3u8_url), video_id, 'Checking for additional m3u8 formats', fatal=False))
+        if ext == 'm3u8':
+            m3u8_fmts, m3u8_subs = self._extract_m3u8_formats_and_subtitles(m3u8_url, video_id, fatal=False)
+            formats.extend(m3u8_fmts)
+            self._merge_subtitles(m3u8_subs, target=subtitles)
```

Comment:
Before this patch, ThePlatform extraction for CBC videos yields only the direct http format, which is why the additional m3u8 formats check is needed.

There are other sites where the m3u8 formats are already extracted initially, and we would be duplicating the formats by extracting them again here.

So we could do something like this to avoid unnecessary re-extraction:
```suggestion
        # With some sites, manifest URL must be forced to extract HLS formats
        if not traverse_obj(formats, lambda _, v: v['format_id'].startswith('hls')):
            m3u8_url = update_url(url, query='mbr=true&manifest=m3u', fragment=None)
            ext = urlhandle_detect_ext(self._request_webpage(
                HEADRequest(m3u8_url), video_id, 'Checking for additional m3u8 formats', fatal=False))
            if ext == 'm3u8':
                m3u8_fmts, m3u8_subs = self._extract_m3u8_formats_and_subtitles(m3u8_url, video_id, fatal=False)
                formats.extend(m3u8_fmts)
                self._merge_subtitles(m3u8_subs, target=subtitles)
```

Reply:
I've tried this and had to revert, I get the error `NameError: name 'traverse_obj' is not defined`. Tbh, I know too little about how lambdas work in Python so I can't really find a way to fix it.
In principle I agree though, there should be a way to avoid duplicate formats. Does it work on your end? I've updated the test so it works again, no geo-restrictions AFAIK and no logon required.

Reply:
need to import it from utils